packages/contracts/test/framework/plugin/plugin-repo.ts

```diff
@@ -61,212 +46,442 @@ describe.skip('PluginRepo', function () {
+      await expect(
+        pluginRepo.updateReleaseMetadata(0, emptyBytes)
+      ).to.be.revertedWithCustomError(pluginRepo, 'ReleaseZeroNotAllowed');
     });
 
-    it('fails if requesting version 0', async () => {
-      const versionIdx = 0;
-      await expect(pluginRepo.getVersionById(versionIdx))
-        .to.be.revertedWithCustomError(pluginRepo, 'VersionIndexDoesNotExist')
-        .withArgs(versionIdx);
+    it('reverts if release does not exist', async () => {
+      await expect(
+        pluginRepo.updateReleaseMetadata(1, emptyBytes)
+      ).to.be.revertedWithCustomError(pluginRepo, 'ReleaseDoesNotExist');
     });
 
-    context('adding new version', () => {
-      let newPluginSetup: string;
-      const newContent = '0x13';
+    it('reverts if metadata length is 0', async () => {
+      await pluginRepo.createVersion(
+        1,
+        pluginSetupMock.address,
+        BUILD_METADATA,
+        RELEASE_METADATA
+      );
+      await expect(
+        pluginRepo.updateReleaseMetadata(1, '0x')
+      ).to.be.revertedWithCustomError(pluginRepo, 'EmptyReleaseMetadata');
+    });
 
-      before(async function () {
-        const pluginSetupMock = await deployMockPluginSetup();
-        newPluginSetup = pluginSetupMock.address;
-      });
+    it('successfuly updates metadata for the release that already exists', async () => {
+      await pluginRepo.createVersion(
+        1,
+        pluginSetupMock.address,
+        BUILD_METADATA,
+        RELEASE_METADATA
+      );
+      await expect(pluginRepo.updateReleaseMetadata(1, '0x11'))
+        .to.emit(pluginRepo, 'ReleaseMetadataUpdated')
+        .withArgs(1, '0x11');
+    });
+  });
 
-      beforeEach(async function () {
-        await pluginRepo.createVersion([2, 0, 0], newPluginSetup, newContent);
-      });
+  describe('Different types of getVersions:', async () => {
+    // R - release, B - build
+    let pluginSetup_R1_B1: PluginUUPSUpgradeableSetupV1Mock;
+    let pluginSetup_R1_B2: PluginUUPSUpgradeableSetupV1Mock;
+    let pluginSetup_R2_B1: PluginUUPSUpgradeableSetupV1Mock;
+    let BUILD_METADATA_R1_B1 = BUILD_METADATA;
+    let BUILD_METADATA_R1_B2 = `${BUILD_METADATA}11`;
+    let BUILD_METADATA_R2_B1 = `${BUILD_METADATA}1111`;
 
-      it('new version is fetchable as latest', async () => {
-        assertVersion(
-          await pluginRepo.getLatestVersion(),
-          [2, 0, 0],
-          newPluginSetup,
-          newContent
-        );
+    beforeEach(async () => {
+      pluginSetup_R1_B1 = pluginSetupMock;
+      pluginSetup_R1_B2 = await deployMockPluginSetup();
+      pluginSetup_R2_B1 = await deployMockPluginSetup();
+
+      await pluginRepo.createVersion(
+        1,
+        pluginSetup_R1_B1.address,
+        BUILD_METADATA_R1_B1,
+        RELEASE_METADATA
+      );
+
+      await pluginRepo.createVersion(
+        1,
+        pluginSetup_R1_B2.address,
+        BUILD_METADATA_R1_B2,
+        RELEASE_METADATA
```

Comment:
Do we need this to be the same? Shouldn't be empty?

Reply:
These tests only test `get` type of functions. So, having them the same is fine. Still didn't want to make it `empty` in order to not make it trick us.